cloud/pkg/common/informers/informers.go

```diff
@@ -17,74 +17,148 @@ limitations under the License.
 package informers
 
 import (
+	"fmt"
 	"sync"
 	"time"
 
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/dynamic/dynamicinformer"
 	k8sinformer "k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
 
+	beehiveContext "github.com/kubeedge/beehive/pkg/core/context"
 	"github.com/kubeedge/kubeedge/cloud/pkg/common/client"
 	"github.com/kubeedge/kubeedge/common/constants"
+	edgescheme "github.com/kubeedge/kubeedge/pkg/client/clientset/versioned/scheme"
 	crdinformers "github.com/kubeedge/kubeedge/pkg/client/informers/externalversions"
 )
 
+var (
+	// KubeScheme is built-in resources scheme for converting group,
+	// version, and kind information to and from Go schemas
+	KubeScheme = scheme.Scheme
+
+	// KubeEdgeScheme is KubeEdge CRD resources scheme for converting
+	// group, version, and kind information to and from Go schemas
+	KubeEdgeScheme = edgescheme.Scheme
+)
+
+// InformerPair include informer and lister for resource
+type InformerPair struct {
+	// GenericLister is a lister that helps list generic resource.
+	// All objects returned here must be treated as read-only.
+	Lister cache.GenericLister
+
+	// Informer provides add and get Indexers ability based on SharedInformer.
+	Informer cache.SharedIndexInformer
+}
+
 type newInformer func() cache.SharedIndexInformer
 
 type KubeEdgeCustomInformer interface {
 	EdgeNode() cache.SharedIndexInformer
 }
 
 type Manager interface {
-	GetK8sInformerFactory() k8sinformer.SharedInformerFactory
-	GetCRDInformerFactory() crdinformers.SharedInformerFactory
-	GetDynamicSharedInformerFactory() dynamicinformer.DynamicSharedInformerFactory
+	// GetKubeInformerFactory return kubernetes built-in resources InformerFactory
+	GetKubeInformerFactory() k8sinformer.SharedInformerFactory
+
+	// GetKubeEdgeInformerFactory return KubeEdge CRD resources InformerFactory
+	GetKubeEdgeInformerFactory() crdinformers.SharedInformerFactory
+
+	// GetDynamicInformerFactory return third-party CRD resources InformerFactory
+	GetDynamicInformerFactory() dynamicinformer.DynamicSharedInformerFactory
+
+	// KubeEdgeCustomInformer return custom informer
 	KubeEdgeCustomInformer
```

Comment:
can delete KubeEdgeCustomInfomer in another PR?

Reply:
For now, we can not delete this KubeEdgeCustomInfomer since some customer informer need filter the resource